### PR TITLE
[Aptos Framework][Staking] Allow pending active validator to leave validator set directly without waiting for epoch change

### PR DESCRIPTION
**Pull request by movekevin**
### Description

Currently the pending_active validator needs to wait for the next epoch to be added to the validator set then leave (and wait 1 more epoch to effectively leave)

### Test Plan
Unit tests

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3604)

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/stake.move [hunk 1]
```diff
@@ -819,21 +824,27 @@ module aptos_framework::stake {
...
-            &mut stake_pool.leave_validator_set_events,
-            LeaveValidatorSetEvent {
-                pool_address,
-            },
-        );
+        // If the validator is still pending_active, directly kick the validator out.
+        let maybe_pending_active_index = find_validator(&validator_set.pending_active, pool_address);
+        if (option::is_some(&maybe_pending_active_index)) {
+            vector::swap_remove(
+                &mut validator_set.pending_active, option::extract(&mut maybe_pending_active_index));
```
Comment (zekun000) [inline aptos-move/framework/aptos-framework/sources/stake.move:831]: you probably want to also calibrate the total joining power?

Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/stake.move:832]: Good catch thanks! I updated the logic to subtract the validator's voting power from the per-epoch increase record. Also updated the newly added unit test to validate.

File: aptos-move/framework/aptos-framework/sources/stake.move [hunk 2]
```diff
@@ -819,21 +825,41 @@ module aptos_framework::stake {
...
+
+            // Decrease the voting power increase as the pending validator's voting power was added when they requested
+            // to join. Now that they changed their mind, their voting power should not affect the joining limit of this
+            // epoch.
+            let validator_stake = (coin::value(&stake_pool.active) as u128);
```
Comment (zekun000) [inline aptos-move/framework/aptos-framework/sources/stake.move:837]: probably should use get_next_epoch_voting_power to be consistent with joining function, there's edge case of pending_inactive if it's unlocked

Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/stake.move]: Makes sense. Updated. Pending active validators could have pending_inactive stake if they left and want to rejoin the validator set.

File: aptos-move/framework/aptos-framework/sources/stake.move [hunk 3]
```diff
@@ -461,6 +462,11 @@ module aptos_framework::stake {
         move_to(account, owner_cap);
     }
 
+    /// Destroy `owner_cap`.
+    public fun destroy_owner_cap(owner_cap: OwnerCapability) {
```
Comment (zekun000) [inline aptos-move/framework/aptos-framework/sources/stake.move:466]: how does this function relate to this change?

**Review by zekun000 — APPROVED** [review]

File: aptos-move/framework/aptos-framework/sources/stake.move (hunk 3 above)
Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/stake.move:466]: Not really related. Just wanted to add this so there's a way to destroy owner cap for StakePool that has been dissolved.

**Review by davidiw — APPROVED** [review]

**Pull request merged**